### ISSUE TITLE
Takes longer to load the content when opening the app 

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#21125]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Open app
2. Tap search
## Expected Result:
Search should load immediately
## Actual Result:
Takes longer up to 3 seconds to load. 
it's because account has a  lot of content, but still, 3000ms is not great.

## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android / native 
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [ ] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 
**Reproducible in staging?:** Needs reproduction (Not able to reproduce for me)
**Reproducible in production?:** needs reproduction 
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

**Expensify/Expensify Issue URL:**
**Issue reported by:** @quinthar 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1686617915136649

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by zanyrenney** [comment]
Cross posted this to expensify open source: https://expensify.slack.com/archives/C01GTK53T8Q/p1687281235653819

**Comment by zanyrenney** [comment]
Not able to reproduce.

**Comment by zanyrenney** [comment]
Have bumped this conversation.

**Comment by hannojg** [comment]
Hey, I am able to reproduce that. I am from Margelo and would like to work on this issue (I am already running a performance audit anyways, and the search flow is part of that)

**Comment by zanyrenney** [comment]
Hey, I am heading OOO shortly, so assigning another bug0 team member to help watch over this issue whilst I am OOO. thank you!

**Comment by sophiepintoraetz** [comment]
Keeping you assigned as per the SO, Zany!

**Comment by sophiepintoraetz** [comment]
@hannojg - do you have a PR you can start whipping up?

**Comment by sophiepintoraetz** [comment]
@zanyrenney - I think you're back on Monday, so I'll unassign myself here. I've bumped @hannojg in [slack](https://expensify.slack.com/archives/C01GTK53T8Q/p1688089140983409?thread_ts=1687281235.653819&cid=C01GTK53T8Q) as well, we're just waiting on the PR to get started.

**Comment by zanyrenney** [comment]
Thanks @sophiepintoraetz

**Comment by hannojg** [comment]
Hey, it just got harder for me to reproduce this reliably, but I have it on my plate for this week (I don't have any good proposal yet). Expect a proposal/PR until end of this week 😊

**Comment by zanyrenney** [comment]
Thanks for the update @hannojg

**Comment by hannojg** [comment]
We just made a bunch of proposals that will help with performance when reading data from onyx and first render times:

https://expensify.slack.com/archives/C01GTK53T8Q/p1688580518417739

I will create issues for those proposals. @zanyrenney I will ping you again once the main issue is up, then we can hold this issue until the other issue is solved.

**Comment by zanyrenney** [comment]
Awesome, thank you @hannojg

Is there any update on the status of that yet? Thanks!

**Comment by hannojg** [comment]
By tomorrow we should have the issue, will let you know then!

**Comment by zanyrenney** [comment]
Awesome, thanks @hannojg

**Comment by zanyrenney** [comment]
Thanks @hannojg - if you are working on this in that issue, should we close this one out? let me know your suggestion please!

**Comment by hannojg** [comment]
I would leave this one open as it better describes the issue, and the issue I mentioned might not be enough to completely get rid of this issue. That's something we have to test once them mentioned issue is solved. So I'd hold this one on hold for the mentioned issue.

**Comment by zanyrenney** [comment]
Okay, that makes sense, thank you @hannojg

**Comment by hannojg** [comment]
The Onyx change is of larger nature and is still actively being worked on here:

- https://github.com/Expensify/react-native-onyx/pull/280

**Comment by zanyrenney** [comment]
Still on hold for this issue: https://github.com/Expensify/react-native-onyx/pull/280

**Comment by zanyrenney** [comment]
There are still updates being made to https://github.com/Expensify/react-native-onyx/pull/280

**Comment by hannojg** [comment]
The PR just got merged and we are working on getting it into the app:

- https://github.com/Expensify/App/pull/23749

**Comment by hannojg** [comment]
The aforementioned PR fixes the issue, as also reported by testers (David). Video:
(Notice that there is no blocking in opening the search page and it opens right away):


https://github.com/Expensify/App/assets/16821682/322177ab-db20-41c0-a93d-bb721d718bb6

**Comment by zanyrenney** [comment]
Hey @hannojg would you mind being a bit more clear please?

> The aforementioned PR fixes the issue, as also reported by testers (David)

Do you mean the linked PR https://github.com/Expensify/App/pull/23749 fixes this issue? or that now that issue is merged, we can resume work on this? Thanks!

**Comment by hannojg** [comment]
It means that the linked PR https://github.com/Expensify/App/pull/23749 fixed the issue, yes! 😊
